libs/assertion/src/source_location.cpp

```diff
@@ -0,0 +1,17 @@
+//  Copyright (c) 2019 Thomas Heller
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/assertion/source_location.hpp>
+
+#include <iostream>
+
+namespace hpx { namespace assertion {
+    std::ostream& operator<<(std::ostream& os, source_location const& loc)
```

Comment:
Do we have to export this function?

Reply:
Apparently only used internally so no need to export it for now

Reply:
It's probably a good idea. I think this class can be nicely reused by a future exception module.